airflow/traces/otel_tracer.py

```diff
@@ -0,0 +1,333 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+import random
+
+from opentelemetry import trace
+from opentelemetry.context import create_key
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.resources import HOST_NAME, SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import Span, Tracer as OpenTelemetryTracer, TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
+from opentelemetry.sdk.trace.id_generator import IdGenerator
+from opentelemetry.trace import Link, NonRecordingSpan, SpanContext, TraceFlags, Tracer
+from opentelemetry.trace.span import INVALID_SPAN_ID, INVALID_TRACE_ID
+
+from airflow.configuration import conf
+from airflow.traces import (
+    TRACEPARENT,
+    TRACESTATE,
+)
+from airflow.traces.utils import (
+    gen_dag_span_id,
+    gen_span_id,
+    gen_trace_id,
+    parse_traceparent,
+    parse_tracestate,
+)
+from airflow.utils.net import get_hostname
+
+log = logging.getLogger(__name__)
+
+_NEXT_ID = create_key("next_id")
+
+
+class OtelTrace:
+    """
+    OpenTelemetry Tracing Class.
+
+    Handles all tracing requirements such as getting the tracer, and starting a new span.
+    When OTEL is enabled, the Trace class will be replaced by this class.
+    """
```

Comment:
I'm surprised this docstring passes the mypy tests.  This appears to violate mypy rule D401 and should maybe be:

```suggestion
    """
    Handle all tracing requirements such as getting the tracer, and starting a new span.
    
    When OTEL is enabled, the Trace class will be replaced by this class.
    """
```

The rule is somewhat opaque and confusing, but essentially a docstring should not tell me what I am looking at, it should tell the class or method what to do.  

"OpenTelemetry Tracing Class." would be dropped because I can tell that by the fact that I am looking at a class called OtelTrace.  And so on.

Reply:
I guess, that's because the paragraph is in multi-line? I'll make modification accordingly.